### DISCUSSION TITLE
gatsby-image taking too long to load

### DISCUSSION THREAD

**Discussion by raghavchandak** [gatsbyjs/gatsby#35926] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I have been working on the following website: [www.atelierdesigner.com](https://atelierdesigner.com/)

The home page of this website contains a grid of a large number of images, where all the images are of really high quality. The problem I am facing is that when the user first opens the website, it takes too long for the images to actually load on the page, sometimes around 5 minutes. I have followed all the guides I found on the net and can't seem to understand where I am going wrong. The code for the home page is as follows:

```
import { graphql } from "gatsby"
import React from "react"
import Layout from "../components/Layout"
import { GatsbyImage, getImage } from "gatsby-plugin-image"
import * as styles from "../styles/home.module.css"

export default function Home({ data }) {
  return (
    <Layout>
      <div className={styles.gallery}>
        {data.allFile.nodes.map(homeimg => {
          if (homeimg.childImageSharp != null) {
            let newimg = getImage(homeimg)
            const imgWidth = homeimg.childImageSharp.gatsbyImageData.width
            const imgHeight = homeimg.childImageSharp.gatsbyImageData.height

            return (
              <GatsbyImage
                image={newimg}
                key={homeimg.childImageSharp.id}
                alt=""
                objectFit="cover"
                className={
                  imgWidth > 4000
                    ? styles.big
                    : imgWidth > 260
                    ? imgHeight == 4032
                      ? styles.separate
                      : styles.small
                    : styles.extra
                }
                objectPosition={imgWidth > 4000 ? "50% 50%" : "50% 0%"}
              />
            )
          }
        })}
      </div>
    </Layout>
  )
}

export const query = graphql`
  query HomeGrid {
    allFile(
      sort: { order: ASC, fields: name }
      filter: { absolutePath: { regex: "/images/home/" } }
    ) {
      nodes {
        childImageSharp {
          gatsbyImageData(
            quality: 100
          )
          id
        }
        id
      }
    }
  }
`
```
I am using the Contenful CMS to store and load my images. The first thing I suspected was that this is happening because of the large size of the images themselves. The images that were uploaded to the CMS were all between 4-5 megabytes in size. However, on inspecting the Network tab of the page, I noticed that all the images were a few hundred kilobytes in size. My guess is that GatsbyImage optimizes the images to reduce them in size.

If the image size is not the issue, I am not sure where I am going wrong and why the page is taking so long to load the images.

Another doubt I had is a conceptual one: If Gatsby is a static site generator, shouldn't the page be displayed immediately with the images? Why do the images need to be loaded seperately?

Kindly help me out. Please let me know if any further detail is required to diagnose the issue.

Any help would be really appreciated. Thank you in advance!

### Reproduction Link

https://github.com/raghavchandak/Atelier-Portfolio

### Steps to Reproduce

You can either clone my github repository or open the website link to see the issue

### Expected Result

The page should be loaded immediately with all the images.

### Actual Result

It takes a really long time for all the images to load.

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
  Binaries:
    Node: 14.15.4 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.7
  Browsers:
    Chrome: 102.0.5005.115
    Edge: Spartan (44.19041.1266.0), Chromium (102.0.1245.33)
  npmPackages:
    gatsby: ^4.13.1 => 4.13.1
    gatsby-image: ^3.11.0 => 3.11.0
    gatsby-plugin-gatsby-cloud: ^4.15.0 => 4.15.0
    gatsby-plugin-image: ^2.13.0 => 2.13.0
    gatsby-plugin-manifest: ^4.15.0 => 4.15.0
    gatsby-plugin-sharp: ^4.13.0 => 4.13.0
    gatsby-source-contentful: ^7.11.0 => 7.11.0
    gatsby-source-filesystem: ^4.13.0 => 4.13.0
    gatsby-transformer-remark: ^5.13.0 => 5.13.0
    gatsby-transformer-sharp: ^4.13.0 => 4.13.0
  npmGlobalPackages:
    gatsby-cli: 4.13.0
```

**Comment by LekoArts** [comment]
Hi!

The images load fine for me in most cases but I'm also on a fast connection. What you see are the placeholders that eventually fade into the images. They take so long because your images are huge. By default images are lazy-loaded so only when we think that you scroll to the image the images are loaded.

You should do multiple things:

1) Update to latest `gatsby` and `gatsby-plugin-version`
2) Use `loading="eager"` for your above the fold images: https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/#shared-props
3) Limit the `quality` to something like `90` or `95`
4) Set a `width` to something reasonable in your query (https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/#widthheight) as you load huge images. Maybe something like `1200px`.
  ![image](https://user-images.githubusercontent.com/16143594/173375716-8a03dcdc-75a1-400a-ae4c-633d9947719d.png)

**Reply by raghavchandak** [reply]
Thanks for the reply! I applied the changes you mentioned and the images load faster now, but still take a while. How can it be reduced further?

One more thing: I am unable to use a fixed width for my images, as doing so totally messes up my grid layout. Is there a workaround for that?
